File: app/src/pepper/java/io/github/anonymous/pepper_realtime/robot/RobotControllerImpl.java
package io.github.anonymous.pepper_realtime.robot;

import com.aldebaran.qi.sdk.QiContext;

/**
 * Pepper-specific implementation of RobotController.
 * Provides access to real QiContext for full robot functionality.
 */
public class RobotControllerImpl implements RobotController {
    
    private QiContext qiContext;
    
    /**
     * Set the QiContext when robot focus is gained
     */
    public void setQiContext(QiContext qiContext) {
        this.qiContext = qiContext;
    }
    
    @Override
    public Object getRobotContext() {
        return qiContext;
    }
    
    @Override
    public boolean isRobotHardwareAvailable() {
        return true;  // Running on Pepper hardware
    }
    
    @Override
    public boolean isReady() {
        return qiContext != null;
    }
    
    @Override
    public String getModeName() {
        return "Pepper Robot Mode";
    }
}

File: app/src/pepper/java/io/github/anonymous/pepper_realtime/robot/RobotControllerImpl.kt
package io.github.anonymous.pepper_realtime.robot

import com.aldebaran.qi.sdk.QiContext

/**
 * Pepper-specific implementation of RobotController.
 * Provides access to real QiContext for full robot functionality.
 */
class RobotControllerImpl : RobotController {

    private var qiContext: QiContext? = null

    /**
     * Set the QiContext when robot focus is gained
     */
    fun setQiContext(qiContext: QiContext?) {
        this.qiContext = qiContext
    }

    override fun getRobotContext(): Any? = qiContext

    override fun isRobotHardwareAvailable(): Boolean = true // Running on Pepper hardware

    override fun isReady(): Boolean = qiContext != null

    override fun getModeName(): String = "Pepper Robot Mode"
}

